### PR TITLE
Adding support in System for hl

### PR DESCRIPTION
**Pull request by Cheemsandfriends**
While I was going through System, I noticed that you couldn't check the totalMemory, so I was searching and I found out that it is totally possible to check the memory, (I'm not sure about gc, but it seems like the same)

### CODE REVIEW

**Comment by loudoweb** [comment]
Thanks for your PR. I've made a similar PR one year ago here https://github.com/openfl/openfl/pull/2491. 
But yours seems more complete because you have added gc call, I haven't tested though.
Why to chose `untyped` instead of `int` for memory?

**Comment by Cheemsandfriends** [comment]
> Why to chose `untyped` instead of `int` for memory?

I was just following what the majority had, which was `untyped`.
But I think it would be better to remove that since `hl.Gc` seems to be pretty close to neko's?

File: src/openfl/system/System.hx [hunk 1]
```diff
@@ -195,6 +197,8 @@ import cpp.vm.Gc;
 	{
 		#if (cpp || neko)
 		return Gc.run(true);
+		#elseif hl
+		return Gc.enable(true);
```
Comment (Apprentice-Alchemist) [inline src/openfl/system/System.hx]: If I remember correctly `Gc.major` is the correct function.
```suggestion
		return Gc.major();
```

File: src/openfl/system/System.hx [hunk 2]
```diff
@@ -262,6 +266,8 @@ import cpp.vm.Gc;
 		#elseif (js && html5)
 		return
 			untyped #if haxe4 js.Syntax.code #else __js__ #end ("(window.performance && window.performance.memory) ? window.performance.memory.usedJSHeapSize : 0");
+		#elseif hl
+		return untyped Gc.stats().currentMemory;
```
Comment (Apprentice-Alchemist) [inline src/openfl/system/System.hx]:
```suggestion
		return Std.int(Gc.stats().currentMemory);
```

**Review by Apprentice-Alchemist — CHANGES_REQUESTED** [review]

**Pull request merged**